cmd/cmd_linux.go

```diff
@@ -79,27 +79,36 @@ func distroSupportsReExec() bool {
 // version of core that do not yet have it.
 func coreSupportsReExec(corePath string) bool {
 	fullInfo := filepath.Join(corePath, filepath.Join(dirs.CoreLibExecDir, "info"))
-	if !osutil.FileExists(fullInfo) {
-		return false
-	}
 	content, err := ioutil.ReadFile(fullInfo)
 	if err != nil {
-		logger.Noticef("cannot read snapd info file %q: %s", fullInfo, err)
+		if !os.IsNotExist(err) {
+			logger.Noticef("cannot open snapd info file %q: %s", fullInfo, err)
+		}
 		return false
 	}
-	ver := regexp.MustCompile("(?m)^VERSION=(.*)$").FindStringSubmatch(string(content))
-	if len(ver) != 2 {
-		logger.Noticef("cannot find snapd version information in %q", content)
-		return false
+
+	if !bytes.HasPrefix(content, []byte("VERSION=")) {
```

Comment:
(nitpick) I think its slightly easier to read if this goes into a helper function (like versionFromInfoContent() or something). The old code was relatively simple (just a regexp match) but the new one requires more state when reading it.